src/tfr_navigation/src/navigation_goal_manager.cpp

```diff
@@ -0,0 +1,61 @@
+#include <navigation_goal_manager.h>
+
+//c++11 is weird with constexpre and we need to define these outside of the
+//class in addition to the declaration and initialization step
+constexpr double NavigationGoalManager::SAFE_MINING_DISTANCE;
+constexpr double NavigationGoalManager::FINISH_LINE;
+constexpr double NavigationGoalManager::MINING_LINE_LENGTH;
+
+NavigationGoalManager::NavigationGoalManager(uint8_t code): location_code{code}
+{
+    initialize_goal();
+}
+NavigationGoalManager::NavigationGoalManager():
+    NavigationGoalManager{tfr_msgs::NavigationGoal::UNSET} {}
+
+
+/**
+ *  Initializes the goal based on the location code, flashes a warning if it's
+ *  not recognized.
+ * */
+void NavigationGoalManager::initialize_goal() {
+    nav_goal.target_pose.header.stamp = ros::Time::now();
+    //TODO integrate the bin reference frame here, upon completion of
+    //the bin transform publisher
+    switch(location_code)
+    {
+        case(tfr_msgs::NavigationGoal::TO_MINING):
+            nav_goal.target_pose.pose.position.x = SAFE_MINING_DISTANCE;
+            //TODO 
```

Comment:
I assume this is the place where we write in the logic for mapping the intermittent path?